tap_freshdesk/test_streams.py

```diff
@@ -0,0 +1,147 @@
+import unittest
+from unittest import mock
+from parameterized import parameterized
+from tap_freshdesk.streams import Agents, Tickets
+
+
+class TestSyncObj(unittest.TestCase):
+    """
+    Test `sync_obj` method of stream.
+    """
+
+    start_date = "2019-06-01T00:00:00Z"
+    only_parent_response = [
+        [{"id": i, "updated_at": f"2020-0{i}-01T00:00:00Z"} for i in [1,5,2]],    # Tickets Response
+        [{"id": "33", "updated_at": f"2020-03-01T00:00:00Z"}],                   # Deleted tickets Response
+        [{"id": "55", "updated_at": f"2020-04-01T00:00:00Z"}],                   # Spam tickets Response
+    ]
+    written_states_1 = {
+        "tickets": "2020-05-01T00:00:00Z",
+        "tickets_deleted": "2020-03-01T00:00:00Z",
+        "tickets_spam": "2020-04-01T00:00:00Z",
+    }
+    with_child_response = [
+        [{"id": i, "updated_at": f"2020-0{i}-01T00:00:00Z"} for i in [1,5,2]],    # Tickets Response
+        [{"id": i, "updated_at": f"2020-0{i}-01T00:00:00Z"} for i in [2,4]],    # conversations Response
+        [{"id": "33", "updated_at": "2020-03-01T00:00:00Z"}],                   # conversations Response
+        [{"id": "55", "updated_at": "2020-04-01T00:00:00Z"}],                   # conversations Response
+        [],[]   # Deleted/Spam tickets response
+    ]
+    written_states_2 = {
+        "conversations": "2020-04-01T00:00:00Z",
+    }
+    written_states_3 = {
+        "tickets": "2020-05-01T00:00:00Z",
+        "conversations": "2020-04-01T00:00:00Z",
+    }
+    expected_state_1 = {
+            "conversations": {"updated_at": "2020-04-01T00:00:00Z"},
+            "tickets": {"updated_at": "2020-03-15T00:00:00Z"},
+            "tickets_deleted": {"updated_at": "2020-05-01T00:00:00Z"},
+            "tickets_spam": {"updated_at": "2020-04-01T00:00:00Z"}
+        }
+    expected_state_2 = {'conversations': {'updated_at': '2020-04-01T00:00:00Z'},
+                        'tickets': {'updated_at': '2019-06-01T00:00:00Z'},
+                        'tickets_deleted': {'updated_at': '2020-05-01T00:00:00Z'},
+                        'tickets_spam': {'updated_at': '2020-04-01T00:00:00Z'}}
+    expected_state_3 = {
+            **expected_state_1,
+            "tickets": {"updated_at": "2020-03-16T00:00:00Z"},
+        }
+
+    @parameterized.expand([
```

Comment:
Add comments like `#["test_name", "selected_streams"......]` to explain each arguments.

Reply:
Added comments to explain arguments.